bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/ThreadPoolManager.java

```diff
@@ -175,6 +186,38 @@ static public ExecutorService getPool(String poolName) {
         return pool;
     }
 
+    /**
+     * Returns an instance of an expression-driven scheduled thread pool service. If it is the first request for the
+     * given pool name, the
+     * instance is newly created.
+     *
+     * @param poolName a short name used to identify the pool, e.g. "discovery"
+     * @return an instance to use
+     */
+    static public ExpressionThreadPoolExecutor getExpressionScheduledPool(String poolName) {
```

Comment:
'getPool()' returns an 'ExecutorService', 'getScheduledPool()' returns an 'ScheduledExecutorService'...

I would expect that 'getExpressionScheduledPool()' returns some 'ExpressionScheduledExecutorService'. Maybe we can introduce some new interface here?